### PR TITLE
ensure carrier initiated cancel also cancels in enroll

### PR DESCRIPTION
**Pull request by saimekala07**
ME-181206072

### CODE REVIEW

File: spec/models/subscribers/policy_terminations_subscriber_spec.rb [hunk 1]
```diff
@@ -200,3 +202,57 @@
...
+    {
+      :resource_instance_uri => "urn:some_thing:policy#policy_id",
+      :hbx_enrollment_ids => JSON.dump(["4321"]),
+      :qualifying_reason => "non_payment",
+      :event_effective_date => termination_date.strftime("%Y%m%d")
+    }
+  end
+  let(:family) {FactoryBot.create(:family, :with_primary_family_member)}
+  let!(:hbx_enrollment) do
+    FactoryBot.create(:hbx_enrollment, effective_on: Date.new(2022,2,1),
```
Comment (markgoho) [inline spec/models/subscribers/policy_terminations_subscriber_spec.rb:246]: are these dates arbitrary?

File: spec/models/subscribers/policy_terminations_subscriber_spec.rb [hunk 2]
```diff
@@ -200,3 +202,57 @@
     end
   end
 end
+
+describe "given cancel event to cancel terminated coverage" do
+  let(:termination_date) { Date.new(2022, 1, 1) }
```
[the remark this answers is not in the document]
Comment (saimekala07) [reply spec/models/subscribers/policy_terminations_subscriber_spec.rb:207]: Testing for the scenario with speciics dates and the expected outcome

**Review by TreyE — CHANGES_REQUESTED** [review]
Please see my comments.

File: app/models/subscribers/policy_terminations_subscriber.rb [hunk 3]
```diff
@@ -29,10 +29,20 @@ def call(event_name, e_start, e_end, msg_id, payload)
           enrollments.each do |en|
             if is_cancel
               Rails.logger.error("PolicyTerminationsSubscriber") { "Found and attempting to process #{en.hbx_id} as cancel" }
+              next if en.coverage_canceled?
               if en.may_cancel_for_non_payment? && qr_uri=='non_payment'
                 en.cancel_for_non_payment!
               elsif en.may_cancel_coverage?
                 en.cancel_coverage!
+              else
```
[the remark this answers is not in the document]
Comment (saimekala07) [reply app/models/subscribers/policy_terminations_subscriber.rb:37]: @TreyE Adding transition from `terminated` to `canceled` for coverage cancel event which creates an unnecessary issue that does not happen frequently and which does not happen through UI and as these are retro cancel and exceptions I'm handing manually by updating state and recording state transitions.

**Review by TreyE — APPROVED** [review]

**Review by kristinmerbach — APPROVED** [review]

**Pull request merged**